modules/javafx.graphics/src/main/native-prism-d3d/hlsl/vsConstants.h

```diff
@@ -39,19 +39,20 @@ struct Light {
 };
 
 // camera
-float4x4    mViewProj   : register (c0);
+float4x4    mViewProj   : register(c0);
 float4      gCameraPos  : register(c4);
 
 
 float4      gReserved5[5] : register(c5);
 
 // lighting
-Light       sLights[5]   : register(c10);
-float4      gAmbinet         : register (c20);
-float4      gAmbinetData[10] : register (c20);
+Light       sLights[5]        : register(c10);
+float4      gLightsNormDir[5] : register(c20);
+float4      gAmbinet          : register(c25);
+float4      gAmbinetData[10]  : register(c25);
 
 // world
-float4x3    mWorld                : register (c30);
-float4x3    mBones[MAX_BONES]     : register (c30);
+float4x3    mWorld            : register(c35);
+float4x3    mBones[MAX_BONES] : register(c35);
 
 float4      gReserved240[16] : register(c240);
```

Comment:
`gReserved240` is now at the wrong location (it should be 245), so if it were ever used it would be a problem. It should be updated to avoid confusion at least.

Reply:
I think the size should be updated to 11 as well (since we probably don't want to go past 256).

Reply:
I wonder why we need `gReserved240` and `gSomethingElse` (in the pixel shader) at all. If they are at the end, they don't need to be reserved I think.
Also, the vertex shader has `gReserved5[5]` that reserves c5 to c9, but the pixel shader does not have anything on c2 and c3, which are also reserved for something.

Another thing I don't understand is why the vertex shader overlaps register definitions for `gAmbinet` and `gAmbinetData[10]`, and `mWorld` and `mBones[MAX_BONES]` (and why 70 of these?).